crates/ln-dlc-node/src/tests/system_tests/mod.rs

```diff
@@ -139,19 +139,29 @@ pub(crate) async fn setup_ln_node(test_dir: &Path, node_name: &str) -> Node {
         listener.local_addr().expect("To get a free local address")
     };
 
-    // todo: the tests are executed in the crates/ln-dlc-node directory, hence the folder will
-    // be created there. but the creation will fail if the .ldk-data/alice/on_chain has not been
-    // created before.
-    Node::new(
-        node_name.to_string(),
-        Network::Regtest,
-        data_dir.as_path(),
-        address,
-        ELECTRS_ORIGIN.to_string(),
-        seed,
-        ephemeral_randomness,
-    )
-    .await
+    if is_coordinator {
```

Comment:
Hahahaha nooooo. Two functions better than a boolean flag? :stuck_out_tongue:

Reply:
With two functions we would have to duplicate so much stuff, so I went for the easier one :D